app/scripts/metamask-controller.js

```diff
@@ -618,6 +624,22 @@ export default class MetamaskController extends EventEmitter {
       this.phishingController.setStalelistRefreshInterval(30 * SECOND);
     }
 
+    ///: BEGIN:ONLY_INCLUDE_IN(blockaid)
+    this.ppomController = new PPOMController({
```

Comment:
As mentioned in the other PR, would an alternate option be to check the blockaid user preference here before creating this instance?

Reply:
We need to inject the controller instance in middleware engine, I do not think that is possible once engine initialisation is completed. That is reason I preferred this approach.

Reply:
Also, I would avoid writing blockaid specific code to enable/disable blockaid in MetamaskController.

Reply:
I appreciate it adds complexity but as we're talking about a user preference, I'd argue that's a client responsibility.

Plus any required client code would just be to detect the preference change and call suitable methods on the controller such as `disable`.

As for the middleware, a future option could be to create a generic `DisableController` which could be constructed with a callback to detect the controller is disabled, a list of method names, and the original controller instance.

It could then use a proxy to detect calls to the provided methods and either throw or pass to the original controller depending on if the feature was disabled. We'd then do all bindings to this wrapper instead of the internal feature controller.

Reply:
`DisableController` is a good idea.

Let me make PreferenceController change in a following PR.